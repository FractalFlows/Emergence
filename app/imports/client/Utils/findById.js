// @flow
import {
  find,
} from 'lodash/fp'

export default function findById(id:number|string, collection:Array<*>):any {
  return find(doc => doc.id === id, collection)
}
