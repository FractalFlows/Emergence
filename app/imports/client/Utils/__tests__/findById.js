import findById from '../findById'

test('should find by id', () => {
  const collection = [
    { id: 'a', thing: 'x'},
    { id: 'b', thing: 't'},
  ]

  expect(findById('a', collection)).toMatchSnapshot()
})

