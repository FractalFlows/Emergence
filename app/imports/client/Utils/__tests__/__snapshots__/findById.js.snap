exports[`test should find by id 1`] = `
Object {
  "id": "a",
  "thing": "x",
}
`;
